From 6574d263631733f0ac6c96dc66d94faaa5b1160a Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Thu, 2 Dec 2010 15:53:50 +0200
Subject: [PATCH 3/4] yali.storage.bootloader:Re-write batch file before un-chroot

post install in grub package use /etc/grub.conf however if installation
is from usb we must re-write conf file before grub install.
---
 yali/storage/bootloader.py |   18 +++++++++++-------
 1 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/yali/storage/bootloader.py b/yali/storage/bootloader.py
index 319d145..6fecfba 100644
--- a/yali/storage/bootloader.py
+++ b/yali/storage/bootloader.py
@@ -252,7 +252,12 @@ class BootLoader(object):
         with open(os.path.join(ctx.consts.target_dir, self._conf), "w") as grubConfFile:
             grubConfFile.write(s)
 
-        self.writeGrubInstallConf()
+        target_conf_dir = os.path.join(ctx.consts.target_dir, "etc")
+        if os.path.exists(target_conf_dir):
+            ctx.logger.debug("Target grub.conf file is writing")
+            self.writeGrubInstallConf(os.path.join(target_conf_dir, "grub.conf"), removableExists=False)
+
+        self.writeGrubInstallConf("/tmp/batch", removableExists=self.removableExists)
         self.appendOtherSystems()
 
     def appendOtherSystems(self):
@@ -384,21 +389,20 @@ class BootLoader(object):
             for device in devices:
                 deviceMapFile.write("(%s)     %s\n" % (get_disk_name(self.storage, device), device.path))
 
-    def writeGrubInstallConf(self):
+    def writeGrubInstallConf(self, path, removableExists=False):
         stage1Devices = get_physical_devices(self.storage, self.storage.devicetree.getDeviceByName(self.device))
         bootDevices = get_physical_devices(self.storage, self.storage.storageset.bootDevice)
 
-        stage1Path = get_partition_name(self.storage, stage1Devices[0], exists=self.removableExists)
-        bootPartitionPath = get_partition_name(self.storage, bootDevices[0], exists=self.removableExists)
+        stage1Path = get_partition_name(self.storage, stage1Devices[0], exists=removableExists)
+        bootPartitionPath = get_partition_name(self.storage, bootDevices[0], exists=removableExists)
 
         batch_template = """root %s
 setup %s
 quit
 """ % (bootPartitionPath, stage1Path)
 
-        ctx.logger.debug("Batch template: %s" % batch_template)
-        file('/tmp/batch','w').write(batch_template)
-        file('/mnt/target/etc/grub.conf','w').write(batch_template)
+        ctx.logger.debug("Writng Batch template to %s:\n%s" % (path, batch_template))
+        file(path,'w').write(batch_template)
 
 
     def install(self):
-- 
1.7.2.2

